orangecontrib/imageanalytics/widgets/owimageviewer.py

```diff
@@ -311,38 +306,59 @@ def sizeHint(self):
 
     @Inputs.data
     def setData(self, data):
-        self.closeContext()
+        if self.image_attr is not None:
+            # Don't store invalid contexts because they will match anything
+            # and crash the widget when they're used.
+            self.closeContext()
         self.clear()
-        self.data = data
-
-        if data is not None:
-            self.image_model.set_domain(data.domain)
-            self.title_model.set_domain(data.domain)
-            im_attr = filter_image_attributes(self.data)
-            self.image_attr = im_attr[0] if im_attr else None
-            self.title_attr = self.title_model[0] if self.title_model else None
+        if data is None:
+            self.commit.now()
+            return
 
-            self.openContext(data)
+        self.image_model.set_domain(data.domain)
+        self.title_model.set_domain(data.domain)
+        if not self.image_model:
+            self.Error.no_image_attr()
+            self.commit.now()
```

Comment:
By "even though not images" you mean that an attribute is chosen (it cannot not be chosen), but it does not refer to images? For instance, one loads zoo and chooses animal name as image name?

I changed `no_images_shown` from `Error` to `Warning` because the widget still "works", it just shows red icons instead of images. In this sense, it makes sense that it outputs the data.

Reply:
If, on the other hand, there are not string variables, the widget errors and outputs no data. I guess this is appropriate.